File: AlgoDS/C++/Two-Sum.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

vector<int> solve(vector<int> arr, int target) 
{
    map<int, int> visited;

    for (size_t i = 0; i < arr.size(); i++)
    {
        visited.insert(pair<int,int>(target-arr[i], arr[i]));
    }

    for (size_t i = 0; i < arr.size(); i++) 
    {
        if (visited.find(arr[i]) != visited.end()) {
            return vector<int>{arr[i], visited.find(arr[i])->second};
        }
    }

    return vector<int>{};
}

File: AlgoDS/Python/TwoSum.py
"""
Approach #1
"""

def solve(arr: list, target: int) -> list:

    visited = {}

    for current in arr:
        temp = target-current

        if temp in visited:
            return [temp, current]
        
        visited[current] = True

    return []





"""
Approach #2
"""

def solve(arr: list, target: int) -> list:
    

    visited = {}

    for current in arr:
        visited[target-current] = current
    
    for current in arr:
        if current in visited:
            return [visited[current], current]
    
    return []
